src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -20,15 +20,48 @@
 
 namespace Stratis.Bitcoin.Features.Miner
 {
+    /// <summary>
+    /// <see cref="PosMinting"/> is used in order to generate new blocks. It involves a sort of lottery, similar to proof-of-work,
+    /// but the chances of winning this lottery is proportional to how many coins you are staking, not on hashing power.
+    /// </summary>
+    /// <remarks>
+    /// Staking is attempted only if the node is fully synchronized and connected to the network.
+    /// If not it will wait  till node is synced. Only transactions that were confirmed at least
+    /// <see cref="PosConsensusOptions.StakeMinConfirmations"/> blocks ago are eligible for staking.
+    /// </remarks>
+    /// <remarks>
+    /// The overall process for "attempting" to mine a PoS block looks like this:
+    /// 1) Create new block with transactions from <see cref="TxMempool"/>
+    /// 2) Get UTXOs (Unspent Transaction Output) that can participate in staking (have suitable depth).
+    /// 3) Split those UTXOs in subsets so each subset contains max of <see cref="utxoStakeDescriptionsPerCoinstakeWorker"/> transactions
+    /// and run CoinstakeWorker for each subset. Splitting is done only to achieve a good level of parallelism.
+    /// 4) CoinstakeWorker will try different versions of a block we are trying to mine (the only difference is the
```

Comment:
again `<see>` for CW, but if we modify 3) as suggested, then it would be `Each task is run inside <see cref="CoinstakeWorker"/>, which will try ...`

however, not sure if we can speak of different versions of a block (that might be misleading due to what version means in bitcoin)

maybe i'd say `will try to find a solution for proof of stake target. This is done by creating a coinstake transaction with each of the available UTXOs combined with all valid timestamps that were not checked.`